src/table/components/head/HeadCellMenu.tsx

```diff
@@ -165,15 +165,20 @@ const HeadCellMenu = ({ column, tabIndex }: HeadCellMenuProps) => {
     if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
   }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
 
+  const chartId = layout.qInfo.qId as string;
+  const buttonId = `sn-table-head-menu-button-${pageColIdx}-${chartId}`;
+  const menuId = `sn-table-head-menu-${pageColIdx}-${chartId}`;
+
   return menuItemGroups.length ? (
     <HeadCellMenuWrapper rightAligned={headTextAlign === 'right'}>
       <StyledMenuIconButton
         isVisible={openListboxDropdown || openMenuDropdown}
         ref={anchorRef}
         size="small"
         tabIndex={tabIndex}
-        id="sn-table-head-menu-button"
-        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
+        className ='sn-table-head-menu-button'
+        id={buttonId}
+        aria-controls={openMenuDropdown ? menuId : undefined}
         aria-expanded={openMenuDropdown ? 'true' : undefined}
         aria-haspopup="true"
```

Comment:
I haven't been able to find an example that uses all three of these. Not sure if it is the best practice. But this is the way the examples in MUIs documentation work. So wont touch this.